service/matching/tasklist/task_list_manager.go

```diff
@@ -727,6 +732,15 @@ func (c *taskListManagerImpl) DescribeTaskList(includeTaskListStatus bool) *type
 		// fallback to im-memory backlog, if failed to get count from db
 		backlogCount = c.taskAckManager.GetBacklogCount()
 	}
+	isolationGroups := c.config.AllIsolationGroups()
+	pollerCounts := c.getRecentPollersByIsolationGroup()
+	isolationGroupMetrics := make(map[string]*types.IsolationGroupMetrics, len(isolationGroups))
+	for _, group := range isolationGroups {
+		isolationGroupMetrics[group] = &types.IsolationGroupMetrics{
+			NewTasksPerSecond: c.qpsTracker.GroupQPS(group),
```

Comment:
nit: I wonder if it matters that much but having a `c.qpsTracker.Snapshot()` would help with accuracy of `NewTasksPerSecond`s in the response. Atomically gathering qps info of all groups would help us avoid future surprises.

Reply:
With the current implementation we can't actually get a snapshot. With AddTask being high QPS and this being just metric collection I wanted to minimize the performance impact. `sync.Map` is lock-free once the values have been written and unless we introduce a lock to serialize the writes we can't actually collect a consistent state.

I think it's fine if these numbers are a little fuzzy.